src/synapse/azext_synapse/custom.py

```diff
@@ -0,0 +1,178 @@
+
+
+# Spark Session
+def list_spark_session_jobs(cmd, client, workspace_name, spark_pool_name, from_index=None, size=None, detailed=True):
+    return client.list(workspace_name, spark_pool_name, from_index, size, detailed)
+
+
+def create_spark_session_job(cmd, client, workspace_name, spark_pool_name, driver_memory, driver_cores,
+                             executor_memory, executor_cores, num_executors, job_name=None, file=None, class_name=None,
+                             args=None, jars=None, files=None, archives=None, conf=None, artifact_id=None,
+                             tags=None, detailed=True):
+    livy_session_request = ExtendedLivySessionRequest(
+        tags=tags, artifact_id=artifact_id,
+        name=job_name, file=file, class_name=class_name, args=args, jars=jars, files=files, archives=archives,
+        conf=conf, driver_memory=driver_memory, driver_cores=driver_cores, executor_memory=executor_memory,
+        executor_cores=executor_cores, num_executors=num_executors)
+    return client.create(workspace_name, spark_pool_name, livy_session_request, detailed)
+
+
+def get_spark_session_job(cmd, client, workspace_name, spark_pool_name, session_id, detailed=True):
+    return client.get(workspace_name, spark_pool_name, session_id, detailed)
+
+
+def cancel_spark_session_job(cmd, client, workspace_name, spark_pool_name, session_id):
+    return client.delete(workspace_name, spark_pool_name, session_id)
+
+
+def reset_timeout(cmd, client, workspace_name, spark_pool_name, session_id):
+    return client.reset_timeout(workspace_name, spark_pool_name, session_id)
+
+
+def list_spark_session_statements(cmd, client, workspace_name, spark_pool_name, session_id):
+    return client.list_statements(workspace_name, spark_pool_name, session_id)
+
+
+def create_spark_session_statement(cmd, client, workspace_name, spark_pool_name, session_id, code, kind):
+    livy_statement_request = LivyStatementRequestBody(code=code, kind=kind)
+    return client.create_statement(workspace_name, spark_pool_name, session_id, livy_statement_request)
+
+
+def get_spark_session_statement(cmd, client, workspace_name, spark_pool_name, session_id, statement_id):
+    return client.get_statement(workspace_name, spark_pool_name, session_id, statement_id)
+
+
+def cancel_spark_session_statement(cmd, client, workspace_name, spark_pool_name, session_id, statement_id):
+    return client.delete_statement(workspace_name, spark_pool_name, session_id, statement_id)
+
+
+def list_workspaces(cmd, client, resource_group_name=None):  # pylint: disable=unused-argument
+    return client.list_by_resource_group(
+        resource_group_name=resource_group_name) if resource_group_name else client.list()
+
+
+def create_workspace(cmd, client, resource_group_name, workspace_name, account_url, file_system, sql_admin_login_user,
+                     sql_admin_login_password, location, tags=None, identity_type="SystemAssigned"):

```

Comment:
>sql_admin_login_password [](start = 21, length = 24)

In the next version, besides letting users pass plain text in this way, we should provider another way to support secure string in order to enhance security.

Reply:
> > sql_admin_login_password [](start = 21, length = 24)
> 
> In the next version, besides letting users pass plain text in this way, we should provider another way to support secure string in order to enhance security.

Good catch, add a work item